frontend/src/stores/gameSocketStore.ts

```diff
@@ -0,0 +1,26 @@
+import type { Game } from '@/types/game.fe';
+import { defineStore } from 'pinia';
+import { io, type Socket } from 'socket.io-client';
+
+export const useGameSocketStore = defineStore('gameSocket', {
+	//  actions == data definitions
+	state: () => ({
+		isInitialized: false,
+		socket: {} as Socket,
+	}),
+	// getters == computed values
+	getters: {},
+	// actions == methods
+	actions: {
+		async initialize() {
+			if (this.isInitialized) {
+				return;
+			}
+			this.isInitialized = true;
+			(this.socket = io('http://localhost:8080/pong', {
+				withCredentials: true,
+			})),
+				console.log('Game Socket is Initialized:', this.isInitialized);
```

Comment:
something weird happened here with formatting..
i think there might have been a comma instead of a semicolon and prettier made it awkward